torchbox/torchbox.com
# Video (YouTube) embed block - cookie handling and sustainability

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1362060693)

> **Note**
> 
> The ticket is for both FE and BE. While this PR handles both, the focus was primarily on the BE. Additional work required on the FE to ensure consistency with overall site design and technical standards.

### Description of Changes Made

This makes changes to how YouTube video embeds are rendered, as follows

- using the the [lite-youtube-embed npm package](https://www.npmjs.com/package/lite-youtube-embed)
- using a [modified OEmbed finder](https://github.com/RealOrangeOne/wagtail-lite-youtube-embed/blob/9d3f22664efa6bd199c82e51a2f7b06c8b71014e/lite_youtube_embed/embed.py) that uses the above. In order for existing YouTube videos to use this modified OEmbed finder, a management command has been created -- `./manage.py update_youtube_embeds`
- adding a cookie consent overlay banner on YouTube videos

The result is that a YouTube video does not load when a Page loads, unless the user decides to load it. We ask them for consent, but they can decide that they don't want us to ask them every time, in which case the cookie consent overlay banner won't be displayed -- but the video still won't load unless they click on the play button.  

### How to Test

- After building the images and spinning up the containers, run `./manage.py update_youtube_embeds` in the `web` container, then run `honcho start`
- Open a Page that has a YouTube embed (e.g. http://127.0.0.1:8000/blog/charity-seo-what-you-need-to-know-right-now/), or add a YouTube embed to any existing Page's body StreamField  
- Interact with the video on the Page, with the ACs in mind. You can use your browsers' dev tools to check for cookies, storage, network, etc.

### Screenshots

<details>
  <summary>Expand to see more</summary>

  ![image](https://github.com/torchbox/torchbox.com/assets/7713776/e161eee5-349c-4186-8adc-127d5ad14dbe)

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on Ubuntu
  - Latest version of Firefox on Ubuntu

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
